projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -195,7 +195,10 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     set paging(value: boolean) {
         this._paging = value;
         this._pipeTrigger++;
-        this.cdr.markForCheck();
+
+        this.cdr.detectChanges();
```

Comment:
Should `detectChanges()` really be called twice here?

Reply:
Yes. At least for the way the grid currently functions. The first detect changes updates the paginator field, which is a ViewChild of the grid (otherwise it will be null in the calculateGridHeight method) and the second one is to actually update the grid once we have changed its metrics due to the paging showing up or hiding. Similar pattern can be seen in the calculateGridSizes method.